pkg/transforms/policy.go

```diff
@@ -37,6 +41,98 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
 	if okns && okpp {
 		node.Properties["_parentPolicy"] = pnamespace + "/" + ppolicy
 	}
+
+	return &PolicyResource{node: node}
+}
+
+// For cert, config, operator policies.
+// This function returns `annotations`, `_isExternal` for `source`,
+// and `severity`, `compliant`, `remediationAction`, `parentPolicyNs`, `parentPolicyName`
+// If `parentPolicyNs`, `parentPolicyName` do not exist, the policy is a stand-alone policy
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	annos := c.GetAnnotations()
+	// This annotation is large and useless
+	delete(annos, "kubectl.kubernetes.io/last-applied-configuration")
+
+	node.Properties["annotations"] = annos
+
+	node.Properties["_isExternal"] = false
+
+	for _, m := range c.GetManagedFields() {
+		if m.Manager == "multicluster-operators-subscription" ||
+			strings.Contains(m.Manager, "argocd") {
+			node.Properties["_isExternal"] = true
+		}
+	}
+
+	typeMeta := metav1.TypeMeta{
+		Kind:       c.GetKind(),
+		APIVersion: c.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node) // add kind, apigroup and version
+
+	node.Properties["compliant"], _, _ = unstructured.NestedString(c.Object, "status", "compliant")
+
+	node.Properties["severity"], _, _ = unstructured.NestedString(c.Object, "spec", "severity")
+
+	node.Properties["remediationAction"], _, _ = unstructured.NestedString(c.Object, "spec", "remediationAction")
+
+	node.Properties["disabled"], _, _ = unstructured.NestedBool(c.Object, "spec", "disabled")
+
+	return node
+}
+
+// For cert, config policies.
+func StandalonePolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	return &PolicyResource{node: getPolicyCommonProperties(c, node)}
+}
+
+func OperatorPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	node = getPolicyCommonProperties(c, node)
+
+	var deploymentAvailable string
+	var upgradeAvailable string
+
+	conditions, _, _ := unstructured.NestedSlice(c.Object, "status", "conditions")
+	for _, condition := range conditions {
+		mapCondition, ok := condition.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		conditionType, found, err := unstructured.NestedString(mapCondition, "type")
+		if !found || err != nil {
+			continue
+		}
+
+		conditionReason, found, err := unstructured.NestedString(mapCondition, "reason")
+		if !found || err != nil {
+			continue
+		}
+
+		if conditionType == "InstallPlanCompliant" {
+			if conditionReason == "InstallPlanRequiresApproval" || conditionReason == "InstallPlanApproved" {
+				upgradeAvailable = "yes"
```

Comment:
Can you explain the reason?

Reply:
@yiraeChristineKim if the condition can only be true or false, then a boolean is the right type. It also makes API interface simpler because you know there can only be two values rather than wonder what the magic values are (e.g. is it `True`, `YES`, `Y`, `yes`, and etc.).

Reply:
frontend side we can set   ```
deploymentAvailable?: 'yes' | 'no'
  upgradeAvailable?: 'yes' | 'no'
```
so no issue. Bool makes it complicated because I have to use another interface and additional type check. We use **worker** for editing the policies but I am trying to avoid any additional calculations because of performance issues in console.  As far as I know, The Stand-alone table is the only one which uses these properties.

Reply:
@yiraeChristineKim that shouldn't add much. Let's try with the bool and if it's an issue, we can go back to this approach.